File: scripts/convert_to_njk.js
const fs = require('fs');

const pages = [
  'index.html', 'pricing.html', 'about.html', 'services.html',
  'how-it-works.html', 'hospitals.html', 'contact.html', 'contact-new.html'
];

function esc(s) { return s.replace(/"/g, '\\"'); }

for (const fname of pages) {
  const html = fs.readFileSync(fname, 'utf8');

  const titleM = html.match(/<title>(.+?)<\/title>/);
  const title = titleM ? titleM[1] : 'Untitled';
  const descM = html.match(/<meta name="description" content="(.+?)"/);
  const description = descM ? descM[1] : '';
  const canonM = html.match(/<link rel="canonical" href="(.+?)"/);
  const canonical = canonM ? canonM[1] : '';

  // Extract page-specific <style> block
  const styleM = html.match(/<style>([\s\S]*?)<\/style>/);
  const pageStyle = styleM ? styleM[1].trim() : '';

  const ogTitleM = html.match(/<meta property="og:title" content="(.+?)"/);
  const ogTitle = ogTitleM ? ogTitleM[1] : '';
  const ogDescM = html.match(/<meta property="og:description" content="(.+?)"/);
  const ogDescription = ogDescM ? ogDescM[1] : '';

  // Hreflang links
  const hreflangs = [...html.matchAll(/<link rel="alternate" hreflang=.+?>/g)];
  const hreflangStr = hreflangs.map(m => m[0]).join('\n');

  // Schema JSON-LD blocks
  const schemas = [...html.matchAll(/<script type="application\/ld\+json">([\s\S]*?)<\/script>/g)];

  // Extract body: between <body> and <footer
  const bodyM = html.match(/<body[^>]*>([\s\S]*?)<footer class="site-footer"/);
  let body = bodyM ? bodyM[1] : '';
  if (body) {
    // Remove nav block
    body = body.replace(/<div class="nav-overlay"[\s\S]*?<\/nav>/m, '');
    // Remove exit popup
    body = body.replace(/<div id="exitPopup">[\s\S]*?exitPopupShown[\s\S]*?<\/script>/m, '');
    // Remove floating buttons
    body = body.replace(/<div class="floating-buttons">[\s\S]*?<\/div>/m, '');
    // Remove mobile bottom bar
    body = body.replace(/<div class="mobile-bottom-bar">[\s\S]*?<\/div>/m, '');
    // Remove inline nav toggle scripts
    body = body.replace(/<script>[\s\S]*?navToggle[\s\S]*?<\/script>/m, '');
    body = body.trim();
  }

  // Build frontmatter
  const fmLines = ['---'];
  fmLines.push('layout: base.njk');
  fmLines.push(`title: "${esc(title)}"`);
  if (description) fmLines.push(`description: "${esc(description)}"`);
  if (canonical) fmLines.push(`canonical: "${canonical}"`);
  if (ogTitle && ogTitle !== title) fmLines.push(`ogTitle: "${esc(ogTitle)}"`);
  if (ogDescription && ogDescription !== description) fmLines.push(`ogDescription: "${esc(ogDescription)}"`);

  if (pageStyle) {
    fmLines.push('pageStyle: |');
    for (const line of pageStyle.split('\n')) fmLines.push(`  ${line}`);
  }

  if (hreflangStr) fmLines.push(`extraHead: "${esc(hreflangStr)}"`);

  if (schemas.length > 0) {
    fmLines.push('schema: |');
    for (const s of schemas) {
      for (const line of s[1].trim().split('\n')) fmLines.push(`  ${line}`);
    }
  }

  fmLines.push('---');
  const output = fmLines.join('\n') + '\n' + body + '\n';

  const outName = fname.replace('.html', '.njk');
  fs.writeFileSync(outName, output, 'utf8');
  console.log(`OK ${fname} -> ${outName}  (${body.length} chars content, ${pageStyle.length} css, ${schemas.length} schemas)`);
}

File: scripts/_oneoff/convert_to_njk.py
"""Convert root HTML pages to Nunjucks (.njk) templates with frontmatter."""
import re, os

pages = [
    'index.html', 'pricing.html', 'about.html', 'services.html',
    'how-it-works.html', 'hospitals.html', 'contact.html', 'contact-new.html'
]

for fname in pages:
    with open(fname, 'r', encoding='utf-8') as f:
        html = f.read()

    title_m = re.search(r'<title>(.+?)</title>', html)
    title = title_m.group(1) if title_m else 'Untitled'

    desc_m = re.search(r'<meta name="description" content="(.+?)"', html)
    description = desc_m.group(1) if desc_m else ''

    canon_m = re.search(r'<link rel="canonical" href="(.+?)"', html)
    canonical = canon_m.group(1) if canon_m else ''

    # Page-specific <style>
    style_m = re.search(r'(<style>.*?</style>)', html, re.DOTALL)
    page_style = ''
    if style_m:
        raw = style_m.group(1)
        page_style = re.sub(r'</?style>', '', raw).strip()

    og_title_m = re.search(r'<meta property="og:title" content="(.+?)"', html)
    og_title = og_title_m.group(1) if og_title_m else ''
    og_desc_m = re.search(r'<meta property="og:description" content="(.+?)"', html)
    og_desc = og_desc_m.group(1) if og_desc_m else ''

    # Hreflang links (index.html)
    hreflangs = re.findall(r'<link rel="alternate" hreflang=.+?>', html)

    # Schema blocks
    schemas = re.findall(r'<script type="application/ld\+json">(.*?)</script>', html, re.DOTALL)

    # Extract body content between nav and footer
    body_match = re.search(r'<body[^>]*>(.*?)<footer class="site-footer"', html, re.DOTALL)
    body_html = ''
    if body_match:
        body_html = body_match.group(1)
        # Remove nav block
        body_html = re.sub(r'<div class="nav-overlay".*?</nav>', '', body_html, flags=re.DOTALL)
        # Remove exit popup
        body_html = re.sub(r'<div id="exitPopup">.*?</div>\s*<script>.*?exitPopup.*?</script>', '', body_html, flags=re.DOTALL)
        # Remove floating buttons
        body_html = re.sub(r'<div class="floating-buttons">.*?</div>', '', body_html, flags=re.DOTALL)
        # Remove mobile bottom bar
        body_html = re.sub(r'<div class="mobile-bottom-bar">.*?</div>', '', body_html, flags=re.DOTALL)
        # Remove inline nav toggle scripts
        body_html = re.sub(r'<script>.*?navToggle.*?</script>', '', body_html, flags=re.DOTALL)
        body_html = body_html.strip()

    # Build frontmatter
    fm = ['---']
    fm.append('layout: base.njk')

    def esc(s):
        return s.replace('"', '\\"').replace('\n', '\\n')

    fm.append(f'title: "{esc(title)}"')
    if description:
        fm.append(f'description: "{esc(description)}"')
    if canonical:
        fm.append(f'canonical: "{canonical}"')
    if og_title and og_title != title:
        fm.append(f'ogTitle: "{esc(og_title)}"')
    if og_desc and og_desc != description:
        fm.append(f'ogDescription: "{esc(og_desc)}"')

    if page_style:
        fm.append('pageStyle: |')
        for line in page_style.split('\n'):
            fm.append(f'  {line}')

    if hreflangs:
        hl_str = ''.join(hreflangs)
        fm.append(f'extraHead: "{esc(hl_str)}"')

    if schemas:
        fm.append('schema: |')
        for s in schemas:
            for line in s.strip().split('\n'):
                fm.append(f'  {line}')

    fm.append('---')
    frontmatter = '\n'.join(fm)
    output = frontmatter + '\n' + body_html + '\n'

    outname = fname.replace('.html', '.njk')
    with open(outname, 'w', encoding='utf-8') as f:
        f.write(output)

    print(f'OK {fname} -> {outname}  ({len(body_html)} chars content, {len(page_style)} css, {len(schemas)} schemas)')
